### ISSUE TITLE
Return found parameters in the same form as input

### ISSUE THREAD

**Issue by twolodzko** [hyperopt/hyperopt#412]
TL;DR `fmin` returns **flattened** dict of conditional parameter values that differs from the input format and the values passed to the `fn` function.

Example
-----------

Let me use a toy example, say that we want to find parameters of linear regression with "random" data:

```python
import numpy as np
import hyperopt
from hyperopt import fmin, tpe, rand, hp

x = np.linspace(-6, 6, 5000)
f = lambda x : 5 + x * 3 + np.random.normal(size=len(x), scale=2)

def opt_func(param):
    yhat = 0
    if param['b0']:
        yhat += param['b0']['b0_coef']
    if param['b1']:
        yhat += param['b1']['b1_coef'] * x
    if param['b2']:
        yhat += param['b2']['b2_coef'] * x**2
    return np.mean((f(x) - yhat)**2)
```

We define conditional parameters in the search space:

```python
search_space = {
    'b0' : hp.choice('b0', [False, { 'b0_coef' : hp.uniform('b0_coef', -50, 50) }]),
    'b1' : hp.choice('b1', [False, { 'b1_coef' : hp.uniform('b1_coef', -50, 50) }]),
    'b2' : hp.choice('b2', [False, { 'b2_coef' : hp.uniform('b2_coef', -50, 50) }]),
}
```

When we sample from it, we get the samples in the similar "conditional" format as input:

```python
hyperopt.pyll.stochastic.sample(search_space)
### {'b0': False, 'b1': {'b1_coef': -27.446828750200137}, 'b2': False}
```

But the output returned by optimizer differs:

```python
fmin(fn=opt_func, space=search_space, algo=rand.suggest, max_evals=500)
### {'b0': 0, 'b1': 1, 'b1_coef': 3.2529256424748496, 'b2': 0}
```

Why this is a problem?
-----------------------------

Well, this means that we need to define **different** function for unpacking the parameters when optimizing and another one for unpacking the "final" parameters.

Solution
----------

Use `hyperopt.space_eval` for the output as a default.
